### PR TITLE
China lake implementation

### PR DESCRIPTION
**Pull request by NotDeirun**
This implements a pump-action grenade launcher and new, projectile based grenade shells for it.
Currently only adds to the code, I will convert the other grenade launchers to the new projectile grenades later and then add the China Lake to the game itself.

### CODE REVIEW

**Comment by NotDeirun** [comment]
Made a mistake with some of the dmi files, let me fix.

File: code/modules/projectiles/projectile/projectilegrenades.dm [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	var/same_turf_hit_chance = 15
+
+obj/item/projectile/bullet/grenade/frag/grenade_effect(target)
+	fragment_explosion(target, range, f_type, f_amount, f_damage, f_step, same_turf_hit_chance)
+
+obj/item/projectile/bullet/grenade/emp
+	var/heavy_emp_range = 3
+	var/light_emp_range = 8
+
+obj/item/projectile/bullet/grenade/emp/grenade_effect(target)
```
Comment (mattroks101) [inline code/modules/projectiles/projectile/projectilegrenades.dm]:
```suggestion
obj/item/projectile/bullet/grenade/emp/grenade_effect(target)
```

**Comment by NotDeirun** [comment]
> 
> 
> This appears to have a lot of copy and paste code from pump shotguns. It might be best if this was just a child of pump shotguns that fired a grenade round, instead of copying and pasting shotgun code.

Since I plan to convert the other grenade launchers after this: Do you think making them all children of pump shotguns, making some of them children of pump shotguns or making them a child of some general grenade launcher would be the best idea?

**Comment by mattroks101** [comment]
All the ones that involve pumping should be children of shotguns, all the ones that have other special mechanics can be their own thing. Whatever the gun shares the most in common with, so if it behaves identical to a shotgun except it shoots a grenade, then make it a shotgun subtype.

**Comment by NotDeirun** [comment]
Sounds like a good idea. I hope this works.

**Comment by mattroks101** [comment]
This PR has conflicts now.
image

**Comment by NotDeirun** [comment]
Those were in from the start - I used old dmi files. I do think I fixed it though, if I take a look at the diffrences between the two files now I only see the sprites I added.

**Comment by NotDeirun** [comment]
How can I resolve that conflict? The only changes I see, after I fixed my earlier mistake, are still just my additions.

**Comment by MadmanMartian** [comment]
You'll need to

 - Extract your icon changes
 - Point towards eris's master branch, and pull to begin rebasing
 - When the DMI conflicts show up, select the Eris's master version to replace your own
 - Once the rebase is complete, re-add your icon changes
 - Point towards your remote's working branch (this branch) and push changes

**Comment by NotDeirun** [comment]
I followed those steps until the last, but when trying to push to my remote I get this error message:
failed to push some refs to 'https://github.com/NotDeirun/CEV-Eris'
To https://github.com/NotDeirun/CEV-Eris
! refs/heads/China_Lake:refs/heads/China_Lake [rejected] (non-fast-forward)
Done
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
'China_Lake' rejected (non-fast-forward)
error: failed to push some refs to 'https://github.com/NotDeirun/CEV-Eris'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

**Comment by MadmanMartian** [comment]
You'll need to force push your changes

**Comment by NotDeirun** [comment]
Thanks for the help, this almost drove me insane.

File: cev_eris.dme [hunk 2]
```diff
@@ -2185,6 +2185,7 @@
 #include "code\modules\projectiles\guns\launcher\crossbow.dm"
 #include "code\modules\projectiles\guns\launcher\grenade_launcher.dm"
 #include "code\modules\projectiles\guns\launcher\pneumatic.dm"
+#include "code\modules\projectiles\guns\launcher\projectile_greande_launcher.dm"
```
Comment (MadmanMartian) [inline cev_eris.dme:2188]: Typo in the filename

Comment (NotDeirun) [reply cev_eris.dme]: Fixed.

**Pull request merged**